torchaudio/functional.py

```diff
@@ -1239,6 +1240,60 @@ def dcshift(
     return output_waveform
 
 
+def overdrive(
+        waveform: Tensor,
+        gain: float = 20,
+        colour: float = 20
+) -> Tensor:
+    r"""Apply a overdrive effect to the audio. Similar to SoX implementation.
+    This effect applies a non linear distortion to the audio signal.
+
+    Args:
+        waveform (Tensor): audio waveform of dimension of `(..., time)`
+        gain (float): desired gain at the boost (or attenuation) in dB
+            Allowed range of values are 0 to 100
+        colour (float):  controls the amount of even harmonic content in the over-driven output
+            Allowed range of values are 0 to 100
+
+    Returns:
+        Tensor: Waveform of dimension of `(..., time)`
+
+    References:
+        http://sox.sourceforge.net/sox.html
+    """
+    actual_shape = waveform.shape
+
+    if len(actual_shape) == 2:
+        waveform = waveform.unsqueeze(0)
+
+    gain = _dB2Linear(gain)
+    colour = colour / 200
+    last_in = torch.zeros(waveform.shape[0], waveform.shape[1])
+    last_out = torch.zeros(waveform.shape[0], waveform.shape[1])
+
+    temp = waveform * gain + colour
+
+    mask1 = temp < -1
+    temp[mask1] = temp[mask1] * 0 - 2 / 3
```

Comment:
You should be able to do something like

```
temp[mask1] = torch.tensor(-2.0 / 3.0, dtype=dtype, device=device)
```

then leave a comment, like

```
# Wrapping the constant with Tensor is required for Torchscript
```

Reply:
updated